### PR TITLE
provide profiled segment list query

### PR DESCRIPTION
**Pull request by mrproliu**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [x] Bug fix
- [x] New feature provided
- [ ] Improve performance

- Related issues
#4104 

### New feature or improvement
- provide GraphQL query: query profiled segment list
- fix dumping thread snapshot error: ProfileTaskExecutionContext#isStartProfileable
- rename snapshot record index name

### CODE REVIEW

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/storage/query/ITraceQueryDAO.java [hunk 1]
```diff
@@ -42,4 +42,12 @@ TraceBrief queryBasicTraces(long startSecondTB, long endSecondTB, long minDurati
...
      */
     List<Span> doFlexibleTraceQuery(String traceId) throws IOException;
+
+    /**
+     * query trace basic info list from appoint segments
+     * @param segmentIdList
+     * @return
+     * @throws IOException
+     */
+    List<BasicTrace> queryBySegmentIdList(List<String> segmentIdList) throws IOException;
```
[the remark this answers is not in the document]
Comment (wu-sheng) [reply oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/storage/query/ITraceQueryDAO.java]: My idea is, you already have task list, you should query spans of profiled segment.

File: apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/profile/ProfileTaskExecutionContext.java [hunk 2]
```diff
@@ -150,7 +150,7 @@ public ProfileTask getTask() {
 
     public boolean isStartProfileable() {
         // check is out of max sampling count check
-        return totalStartedProfilingCount.incrementAndGet() > task.getMaxSamplingCount();
+        return totalStartedProfilingCount.incrementAndGet() <= task.getMaxSamplingCount();
```
[the remark this answers is not in the document]
Comment (mrproliu) [reply apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/profile/ProfileTaskExecutionContext.java]: If `maxSamplingCount` is `2`, and `totalStartedProfilingCount` start with `0`. 
Use `<` to decide `isStartProfileable`, the second time to `incrementAndGet` will return `false`. That's not right.

File: test/e2e/e2e-profile/e2e-profile-test-runner/src/test/java/org/apache/skywalking/e2e/ProfileVerificationITCase.java [hunk 3]
```diff
@@ -78,8 +75,8 @@ public void setUp() {
         //        final String swWebappPort = System.getProperty("sw.webapp.port", "32783");
         final String swWebappPort = System.getProperty("sw.webapp.port", "12800");
         final String instrumentedServiceHost = System.getProperty("service.host", "127.0.0.1");
-        final String instrumentedServicePort = System.getProperty("service.port", "32782");
-        //        final String instrumentedServicePort = System.getProperty("service.port", "9090");
+//        final String instrumentedServicePort = System.getProperty("service.port", "32782");
+        final String instrumentedServicePort = System.getProperty("service.port", "9090");
```
Comment (wu-sheng) [inline test/e2e/e2e-profile/e2e-profile-test-runner/src/test/java/org/apache/skywalking/e2e/ProfileVerificationITCase.java]: What is the difference? Please remove the useless one.

File: test/e2e/e2e-profile/e2e-profile-test-runner/src/test/java/org/apache/skywalking/e2e/ProfileVerificationITCase.java [hunk 4]
```diff
@@ -89,28 +86,6 @@ public void setUp() {
...
-                if (!traces.isEmpty()) {
-                    break;
-                }
-                Thread.sleep(10000L);
-            } catch (Exception ignored) {
-            }
-        }
-
-        // verify basic info
-        verifyServices(minutesAgo);
```
Comment (wu-sheng) [inline test/e2e/e2e-profile/e2e-profile-test-runner/src/test/java/org/apache/skywalking/e2e/ProfileVerificationITCase.java]: You should still send traffic and verify service/instance/endpoint, just no metrics of `service/instance/endpoint`. If no traffic, then there is no trace to be profiled.

**Comment by wu-sheng** [comment]
`javadoc` failure, please recheck.

**Pull request merged**